src/server/main_service.cc

```diff
@@ -686,9 +686,10 @@ void Service::Init(util::AcceptServer* acceptor, std::vector<facade::Listener*>
   pp_.Await([reg](uint32_t index, ProactorBase* pb) { ServerState::Init(index, reg); });
 
   uint32_t shard_num = GetFlag(FLAGS_num_shards);
-  if (shard_num == 0) {
-    shard_num = pp_.size() > 1 ? pp_.size() - 1 : pp_.size();
```

Comment:
@chakaz can you share with me what was the reason for this fix?

Reply:
Sure, when we benchmarked bullmq I did runs with num_shards == threads in addition to num_shards == (threads - 1), and it was consistently better like this.